### PR TITLE
Simplify handling of `SIZEOF_VOID_P`

### PR DESCRIPTION
**Pull request by dag-erling**

### CODE REVIEW

File: expat/buildconf.sh [hunk 1]
```diff
@@ -31,25 +31,4 @@
...
-    local last_line_to_delete=$(( sizeof_void_p_line_number + 1 ))
-    # Note: Avoiding "sed -i" only for macOS portability.
-    local tempfile="$(mktemp)"
-    sed "${first_line_to_delete},${last_line_to_delete}d" "${filename}" > "${tempfile}"
-    mv "${tempfile}" "${filename}"
-}
-
 autoreconf --warnings=all --install --verbose "$@"
-
-patch_expat_config_h_in expat_config.h.in
```
Comment (hartwork) [inline expat/buildconf.sh:55]: @dag-erling dropping this line allows us to prefix the call to `autoreconf` with `exec ` now. Would be cool to be added with the needed CI fixes.

**Review by hartwork — CHANGES_REQUESTED** [review]
@dag-erling very nice in general, but CI issue https://github.com/libexpat/libexpat/actions/runs/8931876245/job/24567000486?pr=870#step:8:42 needs fixing, and one idea about a detail below:

**Comment by dag-erling** [comment]
> @dag-erling very nice in general, but CI issue https://github.com/libexpat/libexpat/actions/runs/8931876245/job/24567000486?pr=870#step:8:42 needs fixing, and one idea about a detail below:

yeah that was stupid of me, I knew it needed changing but forgot...

**Review by hartwork — APPROVED** [review]
@dag-erling thanks for the adjustments, I like it, very nice to get rid of this wart. :+1:  I'll approve in hope for green CI.

**Pull request merged**